src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1413,39 +1413,27 @@ public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousO
         {
             Debug.Assert(!IsConditionalState);
 
-            int receiverSlot = -1;
+            var anonymousType = (NamedTypeSymbol)node.Type;
+            Debug.Assert(anonymousType.IsAnonymousType);
+
             var arguments = node.Arguments;
-            var constructor = node.Constructor;
-            for (int i = 0; i < arguments.Length; i++)
+            var argumentTypes = arguments.SelectAsArray(arg => VisitRvalueWithResult(arg));
+            if (argumentTypes.All(t => !t.IsNull))
             {
-                var argument = arguments[i];
-                TypeSymbolWithAnnotations argumentType = VisitRvalueWithResult(argument);
-                var parameter = constructor.Parameters[i];
-                ReportArgumentWarnings(argument, argumentType, parameter);
-
-                // https://github.com/dotnet/roslyn/issues/24018 node.Declarations includes
-                // explicitly-named properties only. For now, skip expressions
-                // with implicit names. See NullableReferenceTypesTests.AnonymousTypes_05.
-                if (node.Declarations.Length < arguments.Length)
-                {
-                    continue;
-                }
-
-                PropertySymbol property = node.Declarations[i].Property;
-                if (receiverSlot <= 0)
-                {
-                    receiverSlot = GetOrCreateObjectCreationPlaceholderSlot(node);
-                }
+                anonymousType = AnonymousTypeManager.ConstructAnonymousTypeSymbol(anonymousType, argumentTypes);
+            }
 
-                TypeSymbolWithAnnotations propertyType = property.Type;
-                ReportAssignmentWarnings(argument, propertyType, argumentType, useLegacyWarnings: false);
-                TrackNullableStateForAssignment(argument, propertyType, GetOrCreateSlot(property, receiverSlot), argumentType, MakeSlot(argument));
+            int n = arguments.Length;
+            int receiverSlot = GetOrCreateObjectCreationPlaceholderSlot(node);
+            for (int i = 0; i < n; i++)
+            {
+                var argument = arguments[i];
+                var argumentType = argumentTypes[i];
+                var property = AnonymousTypeManager.GetAnonymousTypeProperty(anonymousType, i);
+                TrackNullableStateForAssignment(argument, property.Type, GetOrCreateSlot(property, receiverSlot), argumentType, MakeSlot(argument));
```

Comment:
Is it valid to pass `argumentType` to this method when `argumentType.IsNull == true`

Reply:
No it's not, thanks.

---
In reply to: [259080576](https://github.com/dotnet/roslyn/pull/33562#discussion_r259080576) [](ancestors = 259080576)